### PR TITLE
Add support for CID v1

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR adds support for CID v1

- [x] Update base schemas
- [x] Implement classes to support multihash CID
- [x] Integrate support for CID v1 in core configs
- [x] Integrate support for CID v1 in CLI tools

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by Karrenbelt — APPROVED** [review]
LGTM

**Comment by Karrenbelt** [comment]
failing CIs are an issue, caused by pipenv lock check. We have that in #222 #223 #224 as well. We need to fix this somehow or skip lock check for now

File: aea/configurations/schemas/definitions.json [hunk 1]
```diff
@@ -87,11 +87,11 @@
     },
     "public_id": {
       "type": "string",
-      "pattern": "^[a-zA-Z_][a-zA-Z0-9_]{0,127}/[a-zA-Z_][a-zA-Z0-9_]{0,127}(:(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?)?(:[a-zA-Z_][a-zA-Z0-9]{45})?$"
+      "pattern": "^[a-zA-Z_][a-zA-Z0-9_]{0,127}/[a-zA-Z_][a-zA-Z0-9_]{0,127}(:(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?)?(:((Qm[a-zA-Z0-9]{44})|(ba[a-zA-Z0-9]{57})))?$"
```
Comment (DavidMinarsch) [inline aea/configurations/schemas/definitions.json:90]: Are they guaranteed fixed length?

Comment (angrybayblade) [reply aea/configurations/schemas/definitions.json:90]: Yes

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**